src/core/tests/type_prop/transpose.cpp

```diff
@@ -230,4 +259,162 @@ TEST(type_prop, transpose_with_empty_order) {
 
     EXPECT_EQ(r->get_output_element_type(0), element::f32);
     EXPECT_TRUE(r->get_output_partial_shape(0).same_scheme(PartialShape({300, 1})));
+    EXPECT_EQ(r->get_output_partial_shape(0), (PartialShape{300, 1}));
+}
+
+/** \brief Transpose with order as parameter shape dimensions. */
+TEST(type_prop, transpose_order_as_parameter_shape) {
+    const auto arg = make_shared<v0::Parameter>(element::f32, PartialShape{Dimension(2, 8), Dimension(4, 16), 6});
+
+    const auto param = make_shared<v0::Parameter>(element::i64, PartialShape{2, 0, 1});
+    const auto shape_of = make_shared<v3::ShapeOf>(param);
+    // order after gather [1, 2, 0]
+    const auto gather = make_shared<v1::Gather>(shape_of,
+                                                op::Constant::create(element::i64, {3}, {2, 0, 1}),
+                                                op::Constant::create(element::i64, {}, {0}));
+
+    const auto r = make_shared<v1::Transpose>(arg, gather);
+
+    ASSERT_EQ(r->get_output_element_type(v1::Transpose::ARG_T), element::f32);
```

Comment:
`EXPECT_EQ` can be used to enable shape check, even with output type mismatch.

Reply:
True, will apply when there will be opportunity to change it.